app/models/grouping.rb

```diff
@@ -778,4 +779,11 @@ def use_section_due_date?
       inviter.section.present? &&
       assignment.section_due_dates.present?
   end
+
+  def test_student_grouping_member
+    members = memberships.joins(:user).where(grouping_id: self.id).count
```

Comment:
We care about `accepted_student_memberships` here not all `memberships`

Try starting with:

```
accepted_student_memberships.joins(:user).pluck('users.type')
```

and see if you can find a more efficient way of checking this